File: docs/Claude-Control-Center-main/backend/app/routes/docker.py
from __future__ import annotations

from flask import Blueprint, jsonify, request

from app.services import docker_service

bp = Blueprint("docker", __name__, url_prefix="/api/docker")


@bp.get("/stacks")
def list_stacks():
    return jsonify(docker_service.list_stacks())


@bp.get("/stacks/<name>/detail")
def stack_detail(name: str):
    return jsonify(docker_service.get_stack_detail(name))


@bp.post("/stacks/<name>/start")
def start_stack(name: str):
    return jsonify(docker_service.stack_action(name, "start"))


@bp.post("/stacks/<name>/stop")
def stop_stack(name: str):
    return jsonify(docker_service.stack_action(name, "stop"))


@bp.post("/stacks/<name>/restart")
def restart_stack(name: str):
    return jsonify(docker_service.stack_action(name, "restart"))


@bp.post("/stacks/<name>/redeploy")
def redeploy_stack(name: str):
    return jsonify(docker_service.redeploy_stack(name))


@bp.get("/stacks/<name>/logs")
def stack_logs(name: str):
    lines = request.args.get("lines", 200, type=int)
    return jsonify(docker_service.get_logs(name, lines))

File: docs/Claude-Control-Center-main/frontend/src/api/docker.ts
import { apiFetch } from './client';
import type { DockerStack, DockerService, DockerActionResult, DockerLogsResult } from '../types';

export const fetchDockerStacks = () =>
  apiFetch<DockerStack[]>('/api/docker/stacks');

export const fetchStackDetail = (name: string) =>
  apiFetch<{ services: DockerService[] }>(`/api/docker/stacks/${encodeURIComponent(name)}/detail`);

export const stackAction = (name: string, action: 'start' | 'stop' | 'restart') =>
  apiFetch<DockerActionResult>(`/api/docker/stacks/${encodeURIComponent(name)}/${action}`, {
    method: 'POST',
  });

export const redeployStack = (name: string) =>
  apiFetch<DockerActionResult>(`/api/docker/stacks/${encodeURIComponent(name)}/redeploy`, {
    method: 'POST',
  });

export const fetchStackLogs = (name: string, lines = 200) =>
  apiFetch<DockerLogsResult>(
    `/api/docker/stacks/${encodeURIComponent(name)}/logs?lines=${lines}`
  );
